iFixit/react-commerce
# Filters: Forward when selecting item type

In brief, prevents page reloads when selecting an `Item Type` facet, and keeps the currently selected filters when an `Item Type` facet is selected/deselected.

We now manually create and parse the client's url based on the current Algolia search state. I followed these [docs](https://www.algolia.com/doc/guides/building-search-ui/going-further/routing-urls/js/#seo-friendly-urls), which walk you through the process. This lets us dynamically change the url of the `Item Type` facet links to include the currently selected filters. It also gives us the power to put the item type in the slug of the url, instead of adding it to the query parameters like a normal filter.

Because `Item Type` can now change without a full page reload, we have to update all the places that depended on `Item Type` being constant. This adds a new hook `useDevicePartsItemType()` that gives you the `Item Type` based on the current search state. 

## QA
Do a full smoke test to make sure nothing is broken compared to `main`. The new feature introduced in this pull is

> Prevents page reloads when selecting an Item Type facet, and keeps the currently selected filters when an Item Type facet is selected/deselected.

CC @sterlinghirsh 

Closes #420
